### PR TITLE
fixes problem with order in scope when counting

### PR DESCRIPTION
**Pull request by dickie81**
having order option in model’s scope when performing a count kicked out
this error:

SequelizeDatabaseError: column “xxxx” must appear in the GROUP BY
clause or be used in an aggregate function

### CODE REVIEW

**Comment by janmeier** [comment]
This is not specifically related to scopes.

I guess we can remove the order - I can't come up with a case where you need an order for count (at least not when we already remove the limit) @mickhansen ?

**Comment by mickhansen** [comment]
I don't think `order` makes sense with any kind of aggregate (unless you're selecting other fields, which you definitely aren't in our `count` method).

**Comment by janmeier** [comment]
@dickie81 If you add some general count tests instead this should be ready for merge :)

**Comment by dickie81** [comment]
I have added some extra count tests & also nulled offset as this was causing issues too.

File: test/integration/model.test.js [hunk 1]
```diff
@@ -1665,6 +1666,54 @@ describe(Support.getTestDialectTeaser('Model'), function() {
       });
     });
 
+    describe("options sent to aggregate", function () {
+      var options, aggregateSpy;
+
+      beforeEach(function () {
+        options = { where: ['username = ?', 'user1']};
```
Comment (janmeier) [inline test/integration/model.test.js]: `where: { username: 'user1' }`

File: test/integration/model.test.js [hunk 2]
```diff
@@ -1665,6 +1666,54 @@ describe(Support.getTestDialectTeaser('Model'), function() {
...
+        });
+      });
+
+      it('modifies option "order" by setting it to null', function() {
+        options.order = "username";
+
+        return this.User.count(options).then(function() {
+          var optsArg = aggregateSpy.args[0][2];
+
+          expect(optsArg.order).to.equal(null);
```
Comment (janmeier) [inline test/integration/model.test.js]: `expect(aggregateSpy).to.have.been.calledWith(sinon.match.any, sinon.match.any, sinon.match.object.and.has('order', null))`

Might not exactly be prettier, but I think it convey the assertion better, and it doesn't poke directly into the args array

**Comment by janmeier** [comment]
LGTM, just a couple of small comments on tests

**Comment by dickie81** [comment]
comments appreciated - changes implemented

**Comment by janmeier** [comment]
Thanks! :)

**Pull request merged**